src/libexpr-c/nix_api_expr.h

```diff
@@ -129,7 +129,8 @@ nix_err nix_value_call_multi(
  *
  * This function converts these Values into their final type.
  *
- * @note This function is mainly needed before calling @ref getters, but not for API calls that return a `Value`.
+ * @note This function is mainly needed before calling @ref getters.
+ * All API calls that return a `Value` force thunks automatically.
```

Comment:
This was a mistake in the design, and it already isn't true anymore that all returned `Value`s are forced.

See https://github.com/NixOS/nix/issues/10499 for the reasons why, and https://github.com/NixOS/nix/pull/10537 for a real world counterexample.

We'll revise this doc entirely when fixing that issue.